contracts/src/token/erc20/mod.rs

```diff
@@ -221,57 +222,34 @@ pub trait IErc20 {
     ) -> Result<bool, Error>;
 }
 
-#[external]
-impl IErc20 for Erc20 {
-    fn total_supply(&self) -> U256 {
-        self._total_supply.get()
-    }
-
-    fn balance_of(&self, account: Address) -> U256 {
-        self._balances.get(account)
-    }
-
-    fn transfer(&mut self, to: Address, value: U256) -> Result<bool, Error> {
-        let from = msg::sender();
-        self._transfer(from, to, value)?;
-        Ok(true)
-    }
-
-    fn allowance(&self, owner: Address, spender: Address) -> U256 {
-        self._allowances.get(owner).get(spender)
-    }
-
-    fn approve(
+/// Interface of an [`Erc20`] contract methods that can be used by the
+/// extensions.
+pub trait IErc20Internal {
```

Comment:
This feels a bit awkward, could you explain why we need it?

Reply:
Resolved with #235